src/ETL/dataset.py

```diff
@@ -50,12 +53,16 @@
 duplicates_data_file = data_dir / "duplicates.txt"
 duplicates_data = pd.read_csv(duplicates_data_file, sep="\n", header=None)[0].tolist()
 
+bucket = storage.Client().bucket(TIF_BUCKET)
+temp_dir = tempfile.gettempdir()
+
 
 @memoize
 def generate_bbox_from_paths() -> Dict[Path, BoundingBox]:
+    cloud_tif_paths = [Path(p) for p in get_cloud_tif_list(TIF_BUCKET)]
     return {
         p: BoundingBox.from_path(p)
-        for p in tqdm(tifs_dir.glob("**/*.tif"), desc="Generating BoundingBoxes from paths")
+        for p in tqdm(cloud_tif_paths, desc="Generating BoundingBoxes from paths")
```

Comment:
Change was super clean because we already have a CropHarvest function to get a list of all tifs on Google Cloud